reactioncommerce/reaction
# create `updateProductField` GQL mutation to replace meteor methods

Resolves parts of #5679   
Impact: **minor**  
Type: **feature|refactor**

## Issue
As part of our effort to demeteorize the app, we need to convert remaining meteor methods to GQL mutations / queries.

## Solution
This PR adds the following mutation: `updateProductField`

This PR removes the ` products/updateProductField` & `products/toggleVisibility` meteor methods.

Instead of creating a second mutation to clone the `products/toggleVisibility` meteor method, I've just confirmed that `toggleVisibility` works in this new mutation.


## Breaking changes
Any custom code using any of the above meteor methods will need to be refactored to use the new GQL methods.

## Testing
1. Start the app.
1. See that you can preform the following actions:
- Update any fields on a product
- Toggle the visibility of a product

## TODO
- Update all client code that calls `` to use `updateProductField` mutation instead
- Add `hashProduct` function to new `updateProductField` mutation
- Possibly add `hashProduct` to utils
